org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractHighlightingTest.xtend

```diff
@@ -0,0 +1,171 @@
+import org.eclipse.xtext.ui.XtextProjectHelper
+import org.eclipse.xtext.ui.editor.XtextEditorInfo
+import org.eclipse.xtext.ui.testing.util.IResourcesSetupUtil
+
+import static extension org.eclipse.xtext.ui.testing.util.IResourcesSetupUtil.addNature
+
+/**
+ * @since 2.15
+ */
+abstract class AbstractHighlightingTest extends AbstractEditorTest {
+
+	@Inject XtextEditorInfo editorInfo
+	@Inject extension FileExtensionProvider
+
+	override protected getEditorId() {
+		editorInfo.getEditorId
+	}
+
+	/**
+	 * @param it - The editor's input text. The input text must contain the given <i>text</i>.
+	 *
+	 * @param text - The text that's highlighting style is to test.
+	 *
+	 * @param fontStyle - The expected font style used when highlighting the given <i>text</i>. 
+	 *
+	 * @param foregroundR - The red component of the expected foreground color when highlighting the given <i>text</i>.
+	 *
+	 * @param foregroundG - The green component of the expected foreground color when highlighting the given <i>text</i>.
+	 *
+	 * @param foregroundB - The blue component of the expected foreground color when highlighting the given <i>text</i>.
+	 */
+	def testHighlighting(CharSequence it, String text, int fontStyle, int foregroundR, int foregroundG, int foregroundB) {
+		testHighlighting(text, fontStyle, foregroundR, foregroundG, foregroundB, 255, 255, 255)
+	}
+
+	/**
+	 * @param it - The editor's input text. The input text must contain the given <i>text</i>.
+	 *
+	 * @param text - The text that's highlighting style is to test.
+	 *
+	 * @param fontStyle - The expected font style used when highlighting the given <i>text</i>. 
+	 *
+	 * @param foregroundR - The red component of the expected foreground color when highlighting the given <i>text</i>.
+	 *
+	 * @param foregroundG - The green component of the expected foreground color when highlighting the given <i>text</i>.
+	 *
+	 * @param foregroundB - The blue component of the expected foreground color when highlighting the given <i>text</i>.
+	 *
+	 * @param backgroundR - The red component of the expected background color when highlighting the given <i>text</i>.
+	 *
+	 * @param backgroundG - The green component of the expected background color when highlighting the given <i>text</i>.
+	 *
+	 * @param backgroundB - The blue component of the expected background color when highlighting the given <i>text</i>.
+	 */
+	def testHighlighting(CharSequence it, String text, int fontStyle,
+		int foregroundR, int foregroundG, int foregroundB, 
+		int backgroundR, int backgroundG, int backgroundB) {
+		// given
+		dslFile
+		// when
+		.openInEditor
+		// then
+		.text(text).isHighlightedIn(fontStyle, foregroundR, foregroundG, foregroundB, backgroundR, backgroundG, backgroundB)
+	}
+
+	protected def dslFile(CharSequence content) {
+		val file = IResourcesSetupUtil.createFile(projectName, fileName, fileExtension, content.toString)
+		file.project.addNature(XtextProjectHelper.NATURE_ID)
```

Comment:
This will unnecessarily slow down all the tests I suppose. The project should have the nature already assigned from the outside.

Reply:
Please add at least a check if the nature is already present (or does addNature(..) perform such a short-circuit already?)